data/prepare_dataset_giant_midi.py

```diff
@@ -0,0 +1,122 @@
+import os
+import math
+
+import fortepyan as ff
+from tqdm import tqdm
+from datasets import Value, Dataset, Features, Sequence, DatasetDict, load_dataset
+
+from data.quantizer import MidiQuantizer
+
+
+def process_record(piece: ff.MidiPiece, sequence_len: int, quantizer: MidiQuantizer) -> list[dict]:
+    piece_quantized = quantizer.quantize_piece(piece)
+
+    midi_filename = piece_quantized.source["midi_filename"]
+
+    piece.df["next_start"] = piece.df.start.shift(-1)
+    piece.df["dstart"] = piece.df.next_start - piece.df.start
+    piece.df["dstart"].fillna(0, inplace=True)
+
+    record = []
+
+    for subset, subset_quantized in zip(
+        piece.df.rolling(window=sequence_len, step=sequence_len),
+        piece_quantized.df.rolling(window=sequence_len, step=sequence_len),
+    ):
```

Comment:
```suggestion
n_samples = 1 + (piece.size - sequence_len) // sequence_step
for it in range(n_samples):
    start = it * sequence_step
    finish = start + sequence_len
    part = piece[start:finish]
```

I think we don't need to use pandas rolling methods like this here, and we also should control the window size with a separate parameter, and not have `sequence_step=sequence_len` hardcoded. 

We also shouldn't base this on the piece.df, `MidiPiece` tracks its source when you use `__getitem__` and we should try to make it possible to track every data record we have back to it's original MIDI recording.

Here's what I mean by the source tracking:
```
In [192]: piece[13:123].source
Out[192]: {'start': 13, 'start_time': 1.0390570000000001, 'finish': 123}
```